src/main/kotlin/io/openfuture/state/mapper/WalletMapperImpl.kt

```diff
@@ -0,0 +1,18 @@
+package io.openfuture.state.mapper
+
+import io.openfuture.state.controller.domain.dto.WalletDto
+import io.openfuture.state.model.Wallet
+import org.springframework.stereotype.Component
+
+@Component
+class WalletMapperImpl : WalletMapper {
+    override fun toWalletDto(wallet: Wallet): WalletDto {
```

Comment:
Never.

Instead, you have to use DTO Constructor: `WalletDto(wallet)`.
Or extension functions maximum.

Mappers are from evil. Always.